components/proxy/src/test/kotlin/com/hotels/styx/routing/handlers/PathPrefixRouterTest.kt

```diff
@@ -0,0 +1,256 @@
+            val fooBazFileHandler = factory.build(RoutingObjectReference("foo-baz-file"))
+
+            val router = PathPrefixRouter(listOf(
+                    pair("/foo", fooFileHandler),
+                    pair("/foo/", fooPathHandler),
+                    pair("/foo/bar", fooBarFileHandler),
+                    pair("/foo/bar/", fooBarPathHandler),
+                    pair("/foo/baz", fooBazFileHandler)
+            ))
+
+            router.route(get("/").build()) shouldBe Optional.empty()
+            router.route(get("/foo").build()) shouldBe Optional.of(fooFileHandler)
+            router.route(get("/foo/x").build()) shouldBe Optional.of(fooPathHandler)
+            router.route(get("/foo/").build()) shouldBe Optional.of(fooPathHandler)
+            router.route(get("/foo/bar").build()) shouldBe Optional.of(fooBarFileHandler)
+            router.route(get("/foo/bar/x").build()) shouldBe Optional.of(fooBarPathHandler)
+            router.route(get("/foo/bar/").build()) shouldBe Optional.of(fooBarPathHandler)
+            router.route(get("/foo/baz/").build()) shouldBe Optional.of(fooBazFileHandler)
+            router.route(get("/foo/baz/y").build()) shouldBe Optional.of(fooBazFileHandler)
+        }
+    }
+
+    feature("PathPrefixRouterFactory") {
+
+        val rootHandler = factory.build(listOf(), routingObjectDef("""
+                type: StaticResponseHandler
+                config:
+                  status: 200
+                  content: root
+                """.trimIndent()))
+
+        val fooHandler = factory.build(listOf(), routingObjectDef("""
+                type: StaticResponseHandler
+                config:
+                  status: 200
+                  content: foo
+                """.trimIndent()))
+
+        scenario("Builds a PathPrefixRouter instance") {
+            val routingDef = routingObjectDef("""
+                  type: PathPrefixRouter
+                  config:
+                    routes:
+                      - { prefix: /, destination: root }
+                      - { prefix: /foo/, destination: foo }
+                """.trimIndent())
+
+            val objectStore = StyxObjectStore<RoutingObjectRecord>()
+            objectStore.insert("root", RoutingObjectRecord("X", routingDef.config(), rootHandler))
+            objectStore.insert("foo", RoutingObjectRecord("X", routingDef.config(), fooHandler))
+
+            val factoryContext = HttpHandlerFactory.Context(
+                    mockk(),
+                    objectStore,
+                    RoutingObjectFactory(RouteRefLookup(objectStore)),
+                    mockk(),
+                    mockk(),
+                    false)
+
+            val handler = PathPrefixRouter.Factory().build(listOf(), factoryContext, routingDef);
+
+            handler.handle(HttpRequest.get("/x").build())
+                    .toMono()
+                    ?.block()
```

Comment:
```suggestion
                    .block()
```
`toMono` is a kotlin method with a non-nullable return type

Reply:
Ah okay. Good catch. I'll change them.